app/jobs/scheduled/sync_imap_idle.rb

```diff
@@ -0,0 +1,88 @@
+
+# frozen_string_literal: true
+
+require_dependency 'imap/sync'
+
+module Jobs
+  class SyncImapIdle < ::Jobs::Scheduled
+    LOCK_KEY ||= 'sync_imap_idle'
+
+    every 30.seconds
+    queue :imap_idle
+
+    def execute(args)
+      return if !SiteSetting.enable_imap || !SiteSetting.enable_imap_idle || !$redis.set(LOCK_KEY, 1, ex: 60.seconds, nx: true)
+
+      @running = true
+      @syncs = {}
+      @syncs_lock = Mutex.new
```

Comment:
Personally I am not a fan of using threads, as historically they have been a source of hard to track down bugs and are very hard to get right. Having said that, if this was discussed outside of this PR and this was the conclusion that is OK, but I would like someone else to review it who has more experience such as @SamSaffron.